### PR TITLE
feat!(comp/exporter/self): rename component to clear up ambiguity

### LINKED ISSUE #6354: Rename prometheus.exporter.agent to prometheus.exporter.self
### Request

Rename prometheus.exporter.agent to prometheus.exporter.self. This would be a breaking change and would have to be documented as soon.

### Use case

prometheus.exporter.agent is potentially ambiguous, as it could be mistaken as an exporter for Prometheus Agent. The "agent" term is something we're also moving internally away from; prometheus.exporter.self better represents what the purpose of the component is (an exporter for the process' own metrics).

### PR DESCRIPTION
**Pull request by hainenber**
#### Which issue(s) this PR fixes

Closes #6354 

#### PR Checklist

- [x] CHANGELOG.md updated
- [x] Documentation added
- [x] Tests updated
- [x] Config converters updated

### CODE REVIEW

File: docs/sources/flow/release-notes.md [hunk 1]
```diff
@@ -41,6 +41,11 @@ This functionality is now only available in the main configuration.
 The `disable_high_cardinality_metrics` configuration argument is used by `otelcol.exporter` components such as `otelcol.exporter.otlp`.
 If you need to see high cardinality metrics containing labels such as IP addresses and port numbers, you now have to explicitly set `disable_high_cardinality_metrics` to `false`.
 
+### Breaking change: Rename component `prometheus.exporter.agent` to `prometheus.exporter.self`
+
+The name `prometheus.exporter.agent` is potentially ambiguous and can be misinterpreted as an exporter for Prometheus agent.
+The new name reflects the component's true purpose, that is exporter of process's own metrics
```
Comment (rfratto) [inline docs/sources/flow/release-notes.md:47]:
```suggestion
The name `prometheus.exporter.agent` is potentially ambiguous and can be misinterpreted as an exporter for Prometheus Agent.
The new name reflects the component's true purpose as an exporter of the process's own metrics.
```

cc @clayton-cornell for whether this needs further improvement

**Review by rfratto — APPROVED** [review]
LGTM, thanks! I'll merge following a review/approval from @clayton-cornell

File: docs/sources/flow/reference/components/prometheus.exporter.self.md [hunk 2]
```diff
@@ -1,57 +1,57 @@
 ---
 aliases:
-- /docs/grafana-cloud/monitor-infrastructure/agent/flow/reference/components/prometheus.exporter.agent/
-- /docs/grafana-cloud/send-data/agent/flow/reference/components/prometheus.exporter.agent/
-canonical: https://grafana.com/docs/agent/latest/flow/reference/components/prometheus.exporter.agent/
+- /docs/grafana-cloud/monitor-infrastructure/agent/flow/reference/components/prometheus.exporter.self/
+- /docs/grafana-cloud/send-data/agent/flow/reference/components/prometheus.exporter.self/
```
Comment (clayton-cornell) [inline docs/sources/flow/reference/components/prometheus.exporter.self.md:4]: I'm not sure we need to update these aliases.  According to: https://grafana.com/docs/writers-toolkit/write/front-matter/#aliases the alias should be the original URL not the new URL.

Comment (clayton-cornell) [inline docs/sources/flow/reference/components/prometheus.exporter.self.md:4]:
```suggestion
- /docs/grafana-cloud/monitor-infrastructure/agent/flow/reference/components/prometheus.exporter.agent/
- /docs/grafana-cloud/send-data/agent/flow/reference/components/prometheus.exporter.agent/
./prometheus.exporter.agent/
```

According to: https://grafana.com/docs/writers-toolkit/write/front-matter/#aliases the alias should be the original URL not the new URL.  I'd leave these as they were. and add one new....

We should do a `make docs` to make sure the alias is working as intended.

Comment (rfratto) [reply docs/sources/flow/reference/components/prometheus.exporter.self.md:4]: (For context, `make docs` is expected to be run from within the `docs/` folder; it doesn't work at the root)

**Comment by hainenber** [comment]
Thanks Roberto, thanks Clayton. I've amended the PR with your proposed fixes and executed `make docs`. All good :D

**Review by clayton-cornell — APPROVED** [review]
Looks good tome now too :-)

**Review by tpaschalis — APPROVED** [review]
Thanks for following up on this! 🙌

**Pull request merged**